great_expectations/rule_based_profiler/parameter_builder/regex_pattern_string_parameter_builder.py

```diff
@@ -0,0 +1,193 @@
+        r"/[A-Za-z0-9\.,;:!?()\"'%\-]+/",  # general text
+        r"^ +/",  # leading space
+        r" +/$",  # trailing space
+        r"/https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#()?&//=]*)/",  #  Matching URL (including http(s) protocol)
+        r"/<\/?(?:p|a|b|img)(?: \/)?>/",  # HTML tags
+        r"/(?:25[0-5]|2[0-4]\d|[01]\d{2}|\d{1,2})(?:.(?:25[0-5]|2[0-4]\d|[01]\d{2}|\d{1,2})){3}/",  # IPv4 IP address
+        r"/(?:[A-Fa-f0-9]){0,4}(?: ?:? ?(?:[A-Fa-f0-9]){0,4}){0,7}/",  # IPv6 IP address,
+    }
+
+    def __init__(
+        self,
+        name: str,
+        metric_domain_kwargs: Optional[Union[str, dict]] = None,
+        metric_value_kwargs: Optional[Union[str, dict]] = None,
+        threshold: float = 1.0,
+        candidate_regexes: Optional[Union[str, dict]] = None,
+        data_context: Optional["DataContext"] = None,
+        batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
+    ):
+        """
+        Configure this RegexPatternStringParameterBuilder
+        Args:
+            name: the name of this parameter -- this is user-specified parameter name (from configuration);
+            it is not the fully-qualified parameter name; a fully-qualified parameter name must start with "$parameter."
+            and may contain one or more subsequent parts (e.g., "$parameter.<my_param_from_config>.<metric_name>").
+            threshold: the ratio of values that must match a format string for it to be accepted
+            candidate_regexes: a list of candidate regex strings that will REPLACE the default
+            data_context: DataContext
+            batch_request: specified in ParameterBuilder configuration to get Batch objects for parameter computation.
+        """
+        super().__init__(
+            name=name,
+            data_context=data_context,
+            batch_request=batch_request,
+        )
+
+        self._metric_domain_kwargs = metric_domain_kwargs
+        self._metric_value_kwargs = metric_value_kwargs
+
+        self._threshold = threshold
+
+        if candidate_regexes is not None:
+            self._candidate_regexes = set(candidate_regexes)
+        else:
+            self._candidate_regexes = RegexPatternStringParameterBuilder.CANDIDATE_REGEX
+
+    def _build_parameters(
+        self,
+        parameter_container: ParameterContainer,
+        domain: Domain,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+    ) -> ParameterContainer:
+        """
+        Check the percentage of values matching the REGEX string, and return the best fit, or None if no
+        string exceeds the configured threshold.
+
+        :return: ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and optional details
+        """
+        validator: Optional[Validator] = self.get_validator(
```

Comment:
@Shinnnyshinshin Getting validator should not be needed any more (nor is getting `batch_ids` is needed), because the `ParameterBuilder` class handles this.  Thanks.